lib/message_bus/rack/middleware.rb

```diff
@@ -39,6 +39,7 @@ def initialize(app, config = {})
     @bus = config[:message_bus] || MessageBus
     @connection_manager = MessageBus::ConnectionManager.new(@bus)
     @started_listener = false
+    @client_id_regexp = Regexp.new(@bus.base_route + 'message-bus/(?<client_id>[[:alnum:]]+)')
```

Comment:
I think you need a regex escape here ... base route could be `test.it` ... then this would match `test7it`

Reply:
Ah, thanks.  I've changed it to using a substring as I was also thinking that client_id might have `-` or other characters.  I thought it was slower but looks like it is about the same (or faster even)

```ruby
base_route = "/base/route/";
path_env = "#{base_route}message-bus/clientId";

rgx = Regexp.new(base_route + 'message-bus/(?<client_id>[[:alnum:]]+)');

substr_proc = -> { path_env["#{base_route}message-bus/".length..-1].split("/")[0] };
regex_proc = -> { path_env.match(rgx) { |m| m[:client_id] } };
split_proc = -> { path_env.split("#{base_route}message-bus/")[1].split("/")[0] };

substr_proc.call # gives "clientId"
regex_proc.call # gives "clientId"
split_proc.call # gives "clientId"

Benchmark.ips { |x| x.report("substr") { substr_proc.call }; x.report("regexp") { regex_proc.call }; x.report("split") { split_proc.call } };
```

```
Warming up --------------------------------------
              substr    75.167k i/100ms
              regexp    67.942k i/100ms
               split    65.618k i/100ms
Calculating -------------------------------------
              substr      1.004M (± 8.0%) i/s -      5.036M in   5.048246s
              regexp    801.254k (±13.0%) i/s -      3.941M in   5.013270s
               split    799.437k (± 6.1%) i/s -      4.003M in   5.025920s
```